pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -126,31 +141,60 @@ function readProfileFiles(rootPath: string, defaultsPath: string, lockedPath: st
  * Windows only. Read settings values from registry using schemaObj as a template.
  * @param schemaObj the object used as a template for navigating registry.
  * @param regKey the registry key obtained from nativeReg.openKey().
- * @param regPath the path to the object relative to regKey.
- * @returns undefined, or the registry data as an object.
+ * @returns null, or the registry data as an object.
  */
-function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY, regPath: string[]): any {
-  let regValue;
-  let newObject: any;
+function readRegistryUsingSchema(schemaObj: any, regKey: nativeReg.HKEY): RecursivePartial<settings.Settings>|null {
+  let newObject: RecursivePartial<settings.Settings>|null = null;
 
-  for (const [schemaKey, schemaVal] of Object.entries(schemaObj)) {
-    if (typeof schemaVal === 'object' && !Array.isArray(schemaVal)) {
-      regValue = readRegistryUsingSchema(schemaVal, regKey, regPath.concat(schemaKey));
-    } else {
-      regValue = nativeReg.getValue(regKey, regPath.join('\\'), schemaKey);
-    }
+  const schemaKeys = Object.keys(schemaObj);
+  // ignore case
+  const registryKeys = nativeReg.enumKeyNames(regKey).concat(nativeReg.enumValueNames(regKey)).map(k => k.toLowerCase());
+  const commonKeys = schemaKeys.filter(k => registryKeys.includes(k.toLowerCase()));
 
-    if (typeof regValue !== 'undefined' && regValue !== null) {
-      newObject ??= {};
+  for (const k of commonKeys) {
+    const schemaVal = schemaObj[k];
+    let regValue: any = null;
+
+    if (typeof schemaVal === 'object') {
+      if (!Array.isArray(schemaVal)) {
+        const innerKey = nativeReg.openKey(regKey, k, nativeReg.Access.READ);
+
+        if (!innerKey) {
+          continue;
+        }
+        try {
+          regValue = readRegistryUsingSchema(schemaVal, innerKey);
+          if (regValue && (typeof regValue === 'object') && Object.keys(regValue).length === 0) {
+            // Ignore empty objects
+            regValue = null;
+          }
+        } finally {
+          nativeReg.closeKey(innerKey);
+        }
+      } else {
+        const multiSzValue = nativeReg.queryValueRaw(regKey, k);
+
+        if (multiSzValue) {
+          // Registry value can be a single-string or even a DWORD and parseMultiString will handle it.
+          const arrayValue = nativeReg.parseMultiString(multiSzValue as nativeReg.Value);
+
+          regValue = arrayValue.length ? arrayValue : null;
+        }
+      }
+    } else {
+      regValue = nativeReg.queryValue(regKey, k);
       if (typeof schemaVal === 'boolean') {
         if (typeof regValue === 'number') {
           regValue = regValue !== 0;
         } else {
-          console.debug(`Deployment Profile expected boolean value for ${ regPath.concat(schemaKey) }`);
+          console.debug(`Deployment Profile expected boolean value for key ${ k }`);
```

Comment:
This leads to error messages like _Deployment Profile expected boolean value for key enabled_ which is pretty useless for troubleshooting.

Reply:
That's because I'm not passing the current path into the recursive function, and I can't get it from the open key AFAICT

Reply:
Right, hence we _should_ pass the path in… otherwise the error messages are not useful.

Reply:
Moved to issue #4153